2022-01-28 21:08:23,315 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-28 21:08:23,319 INFO werkzeug MainThread :  * Running on http://172.18.0.5:5000/ (Press CTRL+C to quit)
2022-01-28 21:11:43,746 INFO werkzeug Thread-1 : 172.18.0.1 - - [28/Jan/2022 21:11:43] "POST /login HTTP/1.1" 200 -
2022-01-28 21:11:55,562 INFO werkzeug Thread-2 : 172.18.0.1 - - [28/Jan/2022 21:11:55] "GET /user HTTP/1.1" 200 -
2022-01-28 21:12:40,946 INFO werkzeug Thread-3 : 172.18.0.1 - - [28/Jan/2022 21:12:40] "[31m[1mPATCH /user/claudia.chapman@duke.edu HTTP/1.1[0m" 401 -
2022-01-28 21:12:50,131 INFO werkzeug Thread-4 : 172.18.0.1 - - [28/Jan/2022 21:12:50] "[31m[1mPATCH /user/claudia.chapman@duke.edu HTTP/1.1[0m" 400 -
2022-01-28 21:13:05,546 INFO werkzeug Thread-5 : 172.18.0.1 - - [28/Jan/2022 21:13:05] "PATCH /user/claudia.chapman@duke.edu HTTP/1.1" 200 -
2022-01-28 21:13:24,995 INFO werkzeug Thread-6 : 172.18.0.1 - - [28/Jan/2022 21:13:24] "[31m[1mPATCH /user/claudia.chapman@duke.edu HTTP/1.1[0m" 400 -
2022-01-28 21:13:39,715 INFO werkzeug Thread-7 : 172.18.0.1 - - [28/Jan/2022 21:13:39] "[31m[1mPATCH /user/claudia.chapman@duke.edu HTTP/1.1[0m" 400 -
2022-01-28 21:13:52,069 INFO werkzeug Thread-8 : 172.18.0.1 - - [28/Jan/2022 21:13:52] "[31m[1mPATCH /user/claudia.chapman@duke.edu HTTP/1.1[0m" 400 -
2022-01-28 21:13:55,345 INFO werkzeug Thread-9 : 172.18.0.1 - - [28/Jan/2022 21:13:55] "[31m[1mPATCH /user/claudia.chapman@duke.edu HTTP/1.1[0m" 400 -
2022-01-28 21:14:00,520 INFO werkzeug Thread-10 : 172.18.0.1 - - [28/Jan/2022 21:14:00] "[31m[1mPATCH /user/claudia.chapman@duke.edu HTTP/1.1[0m" 400 -
2022-01-28 21:17:03,056 INFO werkzeug Thread-11 : 172.18.0.1 - - [28/Jan/2022 21:17:03] "POST /login HTTP/1.1" 200 -
2022-01-28 21:17:24,616 INFO werkzeug Thread-12 : 172.18.0.1 - - [28/Jan/2022 21:17:24] "[31m[1mPATCH /student/5 HTTP/1.1[0m" 401 -
2022-01-28 21:17:32,806 INFO werkzeug Thread-13 : 172.18.0.1 - - [28/Jan/2022 21:17:32] "[31m[1mPATCH /student/5 HTTP/1.1[0m" 422 -
2022-01-28 21:20:26,748 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-28 21:20:26,751 INFO werkzeug MainThread :  * Running on http://172.19.0.5:5000/ (Press CTRL+C to quit)
2022-01-28 21:20:33,959 INFO werkzeug Thread-1 : 172.19.0.1 - - [28/Jan/2022 21:20:33] "POST /login HTTP/1.1" 200 -
2022-01-28 21:20:44,847 INFO werkzeug Thread-2 : 172.19.0.1 - - [28/Jan/2022 21:20:44] "PATCH /student/5 HTTP/1.1" 200 -
2022-01-28 21:21:39,493 INFO werkzeug Thread-3 : 172.19.0.1 - - [28/Jan/2022 21:21:39] "PATCH /student/5 HTTP/1.1" 200 -
2022-01-28 21:22:21,857 INFO werkzeug Thread-4 : 172.19.0.1 - - [28/Jan/2022 21:22:21] "[31m[1mPATCH /student/5 HTTP/1.1[0m" 400 -
2022-01-28 21:23:20,677 INFO werkzeug Thread-5 : 172.19.0.1 - - [28/Jan/2022 21:23:20] "PATCH /student/5 HTTP/1.1" 200 -
2022-01-28 22:16:08,897 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-28 22:16:08,900 INFO werkzeug MainThread :  * Running on http://172.24.0.3:5000/ (Press CTRL+C to quit)
2022-01-29 17:38:15,459 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 17:38:15,461 INFO werkzeug MainThread :  * Running on http://172.28.0.4:5000/ (Press CTRL+C to quit)
2022-01-29 17:39:24,905 INFO werkzeug Thread-1 : 172.28.0.1 - - [29/Jan/2022 17:39:24] "POST /login HTTP/1.1" 200 -
2022-01-29 17:39:54,742 INFO werkzeug Thread-2 : 172.28.0.1 - - [29/Jan/2022 17:39:54] "GET /school HTTP/1.1" 200 -
2022-01-29 17:47:11,851 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 17:47:11,854 INFO werkzeug MainThread :  * Running on http://172.28.0.2:5000/ (Press CTRL+C to quit)
2022-01-29 17:47:23,188 ERROR app Thread-1 : Exception on /school [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 335, in schools_get
    return json.dumps({'success':True, "schools": all_schools})
  File "/usr/lib/python3.8/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/usr/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Student is not JSON serializable
2022-01-29 17:47:23,204 INFO werkzeug Thread-1 : 172.28.0.1 - - [29/Jan/2022 17:47:23] "[35m[1mGET /school HTTP/1.1[0m" 500 -
2022-01-29 17:56:21,128 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 17:56:21,131 INFO werkzeug MainThread :  * Running on http://172.28.0.2:5000/ (Press CTRL+C to quit)
2022-01-29 17:56:31,724 ERROR app Thread-1 : Exception on /school [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 334, in schools_get
    all_schools.append(school.as_dict())
  File "/app/models.py", line 33, in as_dict
    main = {c.key: getattr(self, c.key) for c in inspect(self).mapper.column_attrs}
NameError: name 'inspect' is not defined
2022-01-29 17:56:31,747 INFO werkzeug Thread-1 : 172.28.0.1 - - [29/Jan/2022 17:56:31] "[35m[1mGET /school HTTP/1.1[0m" 500 -
2022-01-29 18:03:54,037 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 18:03:54,040 INFO werkzeug MainThread :  * Running on http://172.19.0.3:5000/ (Press CTRL+C to quit)
2022-01-29 18:05:19,428 INFO werkzeug Thread-1 : 172.19.0.1 - - [29/Jan/2022 18:05:19] "GET /school HTTP/1.1" 200 -
2022-01-29 18:06:07,752 INFO werkzeug Thread-2 : 172.19.0.1 - - [29/Jan/2022 18:06:07] "[31m[1mPOST /route HTTP/1.1[0m" 401 -
2022-01-29 18:06:13,743 INFO werkzeug Thread-3 : 172.19.0.1 - - [29/Jan/2022 18:06:13] "POST /login HTTP/1.1" 200 -
2022-01-29 18:06:28,674 INFO werkzeug Thread-4 : 172.19.0.1 - - [29/Jan/2022 18:06:28] "POST /route HTTP/1.1" 200 -
2022-01-29 18:06:38,837 INFO werkzeug Thread-5 : 172.19.0.1 - - [29/Jan/2022 18:06:38] "POST /route HTTP/1.1" 200 -
2022-01-29 18:06:50,203 INFO werkzeug Thread-6 : 172.19.0.1 - - [29/Jan/2022 18:06:50] "POST /route HTTP/1.1" 200 -
2022-01-29 18:07:04,035 INFO werkzeug Thread-7 : 172.19.0.1 - - [29/Jan/2022 18:07:04] "POST /route HTTP/1.1" 200 -
2022-01-29 18:07:16,403 INFO werkzeug Thread-8 : 172.19.0.1 - - [29/Jan/2022 18:07:16] "[31m[1mGET /school HTTP/1.1[0m" 422 -
2022-01-29 18:07:21,398 INFO werkzeug Thread-9 : 172.19.0.1 - - [29/Jan/2022 18:07:21] "GET /school HTTP/1.1" 200 -
2022-01-29 18:07:34,164 INFO werkzeug Thread-10 : 172.19.0.1 - - [29/Jan/2022 18:07:34] "GET /school/2 HTTP/1.1" 200 -
2022-01-29 18:08:44,254 INFO werkzeug Thread-11 : 172.19.0.1 - - [29/Jan/2022 18:08:44] "GET /school/2 HTTP/1.1" 200 -
2022-01-29 18:14:03,061 INFO werkzeug Thread-12 : 172.19.0.1 - - [29/Jan/2022 18:14:03] "GET /school/2 HTTP/1.1" 200 -
2022-01-29 18:21:38,658 INFO werkzeug Thread-13 : 172.19.0.1 - - [29/Jan/2022 18:21:38] "GET /school/2 HTTP/1.1" 200 -
2022-01-29 18:28:10,873 INFO werkzeug Thread-14 : 172.19.0.1 - - [29/Jan/2022 18:28:10] "GET /school/2 HTTP/1.1" 200 -
2022-01-29 18:33:58,949 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 18:33:58,952 INFO werkzeug MainThread :  * Running on http://172.21.0.4:5000/ (Press CTRL+C to quit)
2022-01-29 18:36:43,130 INFO werkzeug Thread-2 : 172.21.0.1 - - [29/Jan/2022 18:36:43] "GET /school/2 HTTP/1.1" 200 -
2022-01-29 18:36:43,136 INFO werkzeug Thread-1 : 172.21.0.1 - - [29/Jan/2022 18:36:43] "GET /school/2 HTTP/1.1" 200 -
2022-01-29 18:55:43,897 INFO werkzeug Thread-3 : 172.21.0.1 - - [29/Jan/2022 18:55:43] "POST /login HTTP/1.1" 200 -
2022-01-29 18:56:33,569 INFO werkzeug Thread-4 : 172.21.0.1 - - [29/Jan/2022 18:56:33] "GET /user/mary.bertoni@duke.edu HTTP/1.1" 200 -
2022-01-29 18:59:50,502 INFO werkzeug Thread-5 : 172.21.0.1 - - [29/Jan/2022 18:59:50] "GET /school/2 HTTP/1.1" 200 -
2022-01-29 19:01:03,939 INFO werkzeug Thread-6 : 172.21.0.1 - - [29/Jan/2022 19:01:03] "[31m[1mPATCH /student/3 HTTP/1.1[0m" 401 -
2022-01-29 19:01:08,453 INFO werkzeug Thread-7 : 172.21.0.1 - - [29/Jan/2022 19:01:08] "POST /login HTTP/1.1" 200 -
2022-01-29 19:01:19,899 INFO werkzeug Thread-8 : 172.21.0.1 - - [29/Jan/2022 19:01:19] "PATCH /student/3 HTTP/1.1" 200 -
2022-01-29 19:01:26,112 INFO werkzeug Thread-9 : 172.21.0.1 - - [29/Jan/2022 19:01:26] "PATCH /student/5 HTTP/1.1" 200 -
2022-01-29 19:02:17,108 INFO werkzeug Thread-10 : 172.21.0.1 - - [29/Jan/2022 19:02:17] "GET /school/2 HTTP/1.1" 200 -
2022-01-29 19:10:12,139 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 19:10:12,142 INFO werkzeug MainThread :  * Running on http://172.25.0.5:5000/ (Press CTRL+C to quit)
2022-01-29 19:15:31,827 INFO werkzeug Thread-1 : 172.25.0.1 - - [29/Jan/2022 19:15:31] "GET /school/2 HTTP/1.1" 200 -
2022-01-29 21:01:32,987 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 21:01:32,990 INFO werkzeug MainThread :  * Running on http://172.26.0.4:5000/ (Press CTRL+C to quit)
2022-01-29 21:32:44,919 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 21:32:44,924 INFO werkzeug MainThread :  * Running on http://172.27.0.2:5000/ (Press CTRL+C to quit)
2022-01-29 21:34:53,324 INFO werkzeug Thread-1 : 172.27.0.1 - - [29/Jan/2022 21:34:53] "POST /login HTTP/1.1" 200 -
2022-01-29 21:35:50,258 INFO werkzeug Thread-2 : 172.27.0.1 - - [29/Jan/2022 21:35:50] "GET /school?search=Trin HTTP/1.1" 200 -
2022-01-29 21:37:48,296 INFO werkzeug Thread-3 : 172.27.0.1 - - [29/Jan/2022 21:37:48] "[31m[1mPOST /school HTTP/1.1[0m" 401 -
2022-01-29 21:38:25,550 INFO werkzeug Thread-4 : 172.27.0.1 - - [29/Jan/2022 21:38:25] "POST /school HTTP/1.1" 200 -
2022-01-29 21:38:32,140 INFO werkzeug Thread-5 : 172.27.0.1 - - [29/Jan/2022 21:38:32] "GET /school?search=Trin HTTP/1.1" 200 -
2022-01-29 21:38:41,040 INFO werkzeug Thread-6 : 172.27.0.1 - - [29/Jan/2022 21:38:41] "GET /school?search=Jeri HTTP/1.1" 200 -
2022-01-29 22:08:51,136 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 22:08:51,151 INFO werkzeug MainThread :  * Running on http://172.28.0.4:5000/ (Press CTRL+C to quit)
2022-01-29 22:35:05,442 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 22:35:05,446 INFO werkzeug MainThread :  * Running on http://172.29.0.3:5000/ (Press CTRL+C to quit)
2022-01-29 22:40:38,405 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 22:40:38,413 INFO werkzeug MainThread :  * Running on http://172.30.0.3:5000/ (Press CTRL+C to quit)
2022-01-29 23:44:19,074 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 23:44:19,080 INFO werkzeug MainThread :  * Running on http://172.18.0.5:5000/ (Press CTRL+C to quit)
2022-01-29 23:46:55,437 INFO werkzeug Thread-1 : 172.18.0.1 - - [29/Jan/2022 23:46:55] "POST /login HTTP/1.1" 200 -
2022-01-29 23:53:49,154 ERROR app Thread-2 : Exception on /student [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 250, in students_get
    students = Student.query.filter(Student.name.contains(search_keyword))
AttributeError: type object 'Student' has no attribute 'name'
2022-01-29 23:53:49,305 INFO werkzeug Thread-2 : 172.18.0.1 - - [29/Jan/2022 23:53:49] "[35m[1mGET /student?search=32 HTTP/1.1[0m" 500 -
2022-01-29 23:55:27,993 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-29 23:55:27,995 INFO werkzeug MainThread :  * Running on http://172.18.0.5:5000/ (Press CTRL+C to quit)
2022-01-29 23:55:33,869 ERROR app Thread-1 : Exception on /student [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 251, in students_get
    students = students + student
TypeError: unsupported operand type(s) for +: 'BaseQuery' and 'BaseQuery'
2022-01-29 23:55:33,883 INFO werkzeug Thread-1 : 172.18.0.1 - - [29/Jan/2022 23:55:33] "[35m[1mGET /student?search=32 HTTP/1.1[0m" 500 -
2022-01-30 00:21:09,601 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-30 00:21:09,607 INFO werkzeug MainThread :  * Running on http://172.18.0.5:5000/ (Press CTRL+C to quit)
2022-01-30 00:21:39,532 INFO werkzeug Thread-1 : 172.18.0.1 - - [30/Jan/2022 00:21:39] "GET /student?search=32 HTTP/1.1" 200 -
2022-01-30 00:21:49,400 ERROR app Thread-2 : Exception on /student [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 253, in students_get
    students.append(stud)
AttributeError: 'NoneType' object has no attribute 'append'
2022-01-30 00:21:49,414 INFO werkzeug Thread-2 : 172.18.0.1 - - [30/Jan/2022 00:21:49] "[35m[1mGET /student?search=Sim HTTP/1.1[0m" 500 -
2022-01-30 00:23:57,081 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-30 00:23:57,083 INFO werkzeug MainThread :  * Running on http://172.18.0.5:5000/ (Press CTRL+C to quit)
2022-01-30 00:24:02,006 INFO werkzeug Thread-1 : 172.18.0.1 - - [30/Jan/2022 00:24:02] "GET /student?search=Sim HTTP/1.1" 200 -
2022-01-30 00:24:08,799 INFO werkzeug Thread-2 : 172.18.0.1 - - [30/Jan/2022 00:24:08] "GET /student?search=32 HTTP/1.1" 200 -
2022-01-30 00:24:15,475 INFO werkzeug Thread-3 : 172.18.0.1 - - [30/Jan/2022 00:24:15] "GET /student?search=a HTTP/1.1" 200 -
2022-01-30 02:06:49,452 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-30 02:06:49,459 INFO werkzeug MainThread :  * Running on http://172.18.0.5:5000/ (Press CTRL+C to quit)
2022-01-31 01:37:13,412 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 01:37:13,417 INFO werkzeug MainThread :  * Running on http://172.19.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 01:41:13,253 INFO werkzeug Thread-2 : 172.19.0.1 - - [31/Jan/2022 01:41:13] "GET / HTTP/1.1" 200 -
2022-01-31 01:41:13,680 INFO werkzeug Thread-1 : 172.19.0.1 - - [31/Jan/2022 01:41:13] "[33mGET /favicon.ico HTTP/1.1[0m" 404 -
2022-01-31 05:49:16,876 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 05:49:16,883 INFO werkzeug MainThread :  * Running on http://172.19.0.4:5000/ (Press CTRL+C to quit)
2022-01-31 05:49:29,531 INFO werkzeug Thread-1 : 172.19.0.1 - - [31/Jan/2022 05:49:29] "POST /login HTTP/1.1" 200 -
2022-01-31 05:49:59,515 ERROR app Thread-2 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 117, in users_get
    base_query = apply_filters(base_query,filter1)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/filters.py", line 235, in apply_filters
    default_model = get_default_model(query)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 141, in get_default_model
    query_models = get_query_models(query).values()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 64, in get_query_models
    models.extend(mapper.class_ for mapper in query._join_entities)
AttributeError: 'BaseQuery' object has no attribute '_join_entities'
2022-01-31 05:49:59,529 INFO werkzeug Thread-2 : 172.19.0.1 - - [31/Jan/2022 05:49:59] "[35m[1mGET /user?name=claudia HTTP/1.1[0m" 500 -
2022-01-31 05:51:50,911 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 05:51:50,914 INFO werkzeug MainThread :  * Running on http://172.19.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 05:52:10,758 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 117, in users_get
    base_query = apply_filters(base_query,filter1)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/filters.py", line 235, in apply_filters
    default_model = get_default_model(query)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 141, in get_default_model
    query_models = get_query_models(query).values()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 64, in get_query_models
    models.extend(mapper.class_ for mapper in query._join_entities)
AttributeError: 'BaseQuery' object has no attribute '_join_entities'
2022-01-31 05:52:10,770 INFO werkzeug Thread-1 : 172.19.0.1 - - [31/Jan/2022 05:52:10] "[35m[1mGET /user?name=claudia HTTP/1.1[0m" 500 -
2022-01-31 06:00:45,887 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 06:00:45,890 INFO werkzeug MainThread :  * Running on http://172.20.0.5:5000/ (Press CTRL+C to quit)
2022-01-31 06:00:50,747 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 117, in users_get
    base_query = apply_filters(base_query,filter1)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/filters.py", line 235, in apply_filters
    default_model = get_default_model(query)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 141, in get_default_model
    query_models = get_query_models(query).values()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 64, in get_query_models
    models.extend(mapper.class_ for mapper in query._join_entities)
AttributeError: 'BaseQuery' object has no attribute '_join_entities'
2022-01-31 06:00:50,779 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 06:00:50] "[35m[1mGET /user?name=claudia HTTP/1.1[0m" 500 -
2022-01-31 06:07:04,126 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 06:07:04,130 INFO werkzeug MainThread :  * Running on http://172.20.0.4:5000/ (Press CTRL+C to quit)
2022-01-31 06:07:33,255 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 110, in users_get
    base_query = User.Query()
AttributeError: type object 'User' has no attribute 'Query'
2022-01-31 06:07:33,281 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 06:07:33] "[35m[1mGET /user?name=claudia HTTP/1.1[0m" 500 -
2022-01-31 06:08:30,872 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 06:08:30,874 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 06:08:44,353 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 110, in users_get
    base_query = User.query()
TypeError: 'BaseQuery' object is not callable
2022-01-31 06:08:44,365 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 06:08:44] "[35m[1mGET /user?name=claudia HTTP/1.1[0m" 500 -
2022-01-31 06:16:01,902 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 06:16:01,904 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 06:16:18,946 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 118, in users_get
    base_query = apply_filters(base_query,filter1)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/filters.py", line 235, in apply_filters
    default_model = get_default_model(query)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 141, in get_default_model
    query_models = get_query_models(query).values()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 64, in get_query_models
    models.extend(mapper.class_ for mapper in query._join_entities)
AttributeError: 'Query' object has no attribute '_join_entities'
2022-01-31 06:16:18,963 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 06:16:18] "[35m[1mGET /user?name=claudia HTTP/1.1[0m" 500 -
2022-01-31 06:18:24,424 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 06:18:24,426 INFO werkzeug MainThread :  * Running on http://172.20.0.4:5000/ (Press CTRL+C to quit)
2022-01-31 06:18:26,976 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 119, in users_get
    base_query = apply_filters(base_query,filter1)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/filters.py", line 235, in apply_filters
    default_model = get_default_model(query)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 141, in get_default_model
    query_models = get_query_models(query).values()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 64, in get_query_models
    models.extend(mapper.class_ for mapper in query._join_entities)
AttributeError: 'BaseQuery' object has no attribute '_join_entities'
2022-01-31 06:18:26,989 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 06:18:26] "[35m[1mGET /user?name=claudia HTTP/1.1[0m" 500 -
2022-01-31 06:26:20,846 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 06:26:20,852 INFO werkzeug MainThread :  * Running on http://172.20.0.4:5000/ (Press CTRL+C to quit)
2022-01-31 06:26:23,535 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 119, in users_get
    base_query = apply_filters(base_query,filter1)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/filters.py", line 235, in apply_filters
    default_model = get_default_model(query)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 141, in get_default_model
    query_models = get_query_models(query).values()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy_filters/models.py", line 64, in get_query_models
    models.extend(mapper.class_ for mapper in query._join_entities)
AttributeError: 'Query' object has no attribute '_join_entities'
2022-01-31 06:26:23,566 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 06:26:23] "[35m[1mGET /user?name=claudia HTTP/1.1[0m" 500 -
2022-01-31 06:43:32,995 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 06:43:32,997 INFO werkzeug MainThread :  * Running on http://172.20.0.4:5000/ (Press CTRL+C to quit)
2022-01-31 06:45:07,659 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 06:45:07,662 INFO werkzeug MainThread :  * Running on http://172.20.0.4:5000/ (Press CTRL+C to quit)
2022-01-31 06:57:35,052 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 06:57:35,056 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:00:25,225 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:00:25,227 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:00:37,164 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:00:37] "GET /user?name=claudia HTTP/1.1" 200 -
2022-01-31 07:01:48,000 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:01:48] "GET /user?name=claudia HTTP/1.1" 200 -
2022-01-31 07:02:04,912 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:02:04,915 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:02:12,493 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:02:12] "GET /user?name=claudia HTTP/1.1" 200 -
2022-01-31 07:02:49,272 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:02:49,276 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:02:52,529 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:02:52] "GET /user?name=claudia HTTP/1.1" 200 -
2022-01-31 07:03:43,065 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:03:43,067 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:05:14,804 DEBUG root Thread-1 : claudia
2022-01-31 07:05:14,821 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:05:14] "GET /user?name=claudia HTTP/1.1" 200 -
2022-01-31 07:05:33,987 DEBUG root Thread-2 : Claudia
2022-01-31 07:05:33,993 ERROR app Thread-2 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 152, in users_get
    all_users.append(user.as_dict())
  File "/app/models.py", line 25, in as_dict
    students = [student.id for student in self.students]
AttributeError: 'User' object has no attribute 'students'
2022-01-31 07:05:34,057 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:05:34] "[35m[1mGET /user?name=Claudia HTTP/1.1[0m" 500 -
2022-01-31 07:07:44,872 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:07:44,873 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:08:01,251 DEBUG root Thread-1 : Claudia
2022-01-31 07:08:01,291 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:08:01] "GET /user?name=Claudia HTTP/1.1" 200 -
2022-01-31 07:08:49,565 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:08:49,567 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:08:52,127 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 113, in users_get
    filter1 = UserFilter(data={'full_name': name_search, 'email': email})
NameError: name 'email' is not defined
2022-01-31 07:08:52,140 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:08:52] "[35m[1mGET /user?name=Claudia HTTP/1.1[0m" 500 -
2022-01-31 07:09:38,691 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:09:38,693 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:10:10,685 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:10:10,686 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:10:13,063 DEBUG root Thread-1 : Claudia
2022-01-31 07:10:13,098 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:10:13] "GET /user?name=Claudia HTTP/1.1" 200 -
2022-01-31 07:11:01,026 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:11:01,028 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:11:18,901 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:11:18] "GET /user HTTP/1.1" 200 -
2022-01-31 07:12:13,550 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:12:13,560 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:12:14,986 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:12:14] "GET /user HTTP/1.1" 200 -
2022-01-31 07:12:21,629 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:12:21] "GET /user?name=Cla HTTP/1.1" 200 -
2022-01-31 07:12:25,718 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 07:12:25] "GET /user?name=Claud HTTP/1.1" 200 -
2022-01-31 07:12:30,926 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 07:12:30] "GET /user?name=Claud&email=cl HTTP/1.1" 200 -
2022-01-31 07:12:41,938 INFO werkzeug Thread-5 : 172.20.0.1 - - [31/Jan/2022 07:12:41] "GET /user?name=Claud&email HTTP/1.1" 200 -
2022-01-31 07:12:45,261 INFO werkzeug Thread-6 : 172.20.0.1 - - [31/Jan/2022 07:12:45] "GET /user?name=Claud HTTP/1.1" 200 -
2022-01-31 07:13:21,314 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:13:21,317 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:13:22,353 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:13:22] "GET /user?name=Claud HTTP/1.1" 200 -
2022-01-31 07:13:27,510 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:13:27] "GET /user?name=Claud&email HTTP/1.1" 200 -
2022-01-31 07:13:33,807 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 07:13:33] "GET /user?name=Claud&email=duke HTTP/1.1" 200 -
2022-01-31 07:13:38,007 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 07:13:38] "GET /user?name=Claud&email HTTP/1.1" 200 -
2022-01-31 07:13:40,748 INFO werkzeug Thread-5 : 172.20.0.1 - - [31/Jan/2022 07:13:40] "GET /user?name=Claud HTTP/1.1" 200 -
2022-01-31 07:13:46,237 INFO werkzeug Thread-6 : 172.20.0.1 - - [31/Jan/2022 07:13:46] "GET /user HTTP/1.1" 200 -
2022-01-31 07:19:22,026 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:19:22,028 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:19:23,286 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:19:23] "GET /user?sort=full_name HTTP/1.1" 200 -
2022-01-31 07:19:33,362 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:19:33] "GET /user?sort=email HTTP/1.1" 200 -
2022-01-31 07:19:52,092 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 07:19:52] "GET /user?sort=email&name=Cla HTTP/1.1" 200 -
2022-01-31 07:20:02,234 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 07:20:02] "GET /user?email=cl&name=Cla HTTP/1.1" 200 -
2022-01-31 07:20:15,648 INFO werkzeug Thread-5 : 172.20.0.1 - - [31/Jan/2022 07:20:15] "GET /user?email=cl&name=Cla&sort=full_name HTTP/1.1" 200 -
2022-01-31 07:21:44,035 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:21:44,037 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:21:59,054 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 113, in users_get
    if direction.equals('desc'):
AttributeError: 'NoneType' object has no attribute 'equals'
2022-01-31 07:21:59,071 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:21:59] "[35m[1mGET /user?email=cl&name=Cla&sort=full_name&direction=desc HTTP/1.1[0m" 500 -
2022-01-31 07:22:23,267 ERROR app Thread-2 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 113, in users_get
    if direction.equals('desc'):
AttributeError: 'str' object has no attribute 'equals'
2022-01-31 07:22:23,270 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:22:23] "[35m[1mGET /user?email=cl&name=Cla&sort=full_name&dir=desc HTTP/1.1[0m" 500 -
2022-01-31 07:23:09,528 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:23:09,529 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:23:12,332 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:23:12] "GET /user?email=cl&name=Cla&sort=full_name&dir=desc HTTP/1.1" 200 -
2022-01-31 07:23:18,567 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:23:18] "GET /user?email=cl&name=Cla&sort=full_name&dir=asc HTTP/1.1" 200 -
2022-01-31 07:23:23,501 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 07:23:23] "GET /user?email=cl&name=Cla&sort=full_name&dir=desc HTTP/1.1" 200 -
2022-01-31 07:23:29,960 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 07:23:29] "GET /user?email=cl&name=Cla&sort=full_name HTTP/1.1" 200 -
2022-01-31 07:27:07,344 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:27:07,346 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:28:31,977 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:28:31,980 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:28:44,842 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 116, in users_get
    base_query = paginator.get_objects()
NameError: name 'paginator' is not defined
2022-01-31 07:28:44,863 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:28:44] "[35m[1mGET /user?email=cl&name=Cla&sort=full_name&page=1 HTTP/1.1[0m" 500 -
2022-01-31 07:29:16,359 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:29:16,361 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:29:30,969 ERROR app Thread-1 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 122, in users_get
    users = base_query.all()
AttributeError: 'list' object has no attribute 'all'
2022-01-31 07:29:30,977 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:29:30] "[35m[1mGET /user?email=cl&name=Cla&sort=full_name&page=1 HTTP/1.1[0m" 500 -
2022-01-31 07:30:05,178 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:30:05,180 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:30:19,536 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:30:19] "GET /user?email=cl&name=Cla&sort=full_name&page=1 HTTP/1.1" 200 -
2022-01-31 07:30:56,800 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:30:56] "GET /user?sort=full_name&page=1 HTTP/1.1" 200 -
2022-01-31 07:31:08,631 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 07:31:08] "GET /user?sort=full_name&page=2 HTTP/1.1" 200 -
2022-01-31 07:31:11,898 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 07:31:11] "GET /user?sort=full_name&page=3 HTTP/1.1" 200 -
2022-01-31 07:31:14,627 INFO werkzeug Thread-5 : 172.20.0.1 - - [31/Jan/2022 07:31:14] "GET /user?sort=full_name&page=4 HTTP/1.1" 200 -
2022-01-31 07:31:18,014 INFO werkzeug Thread-6 : 172.20.0.1 - - [31/Jan/2022 07:31:18] "GET /user?sort=full_name&page=1 HTTP/1.1" 200 -
2022-01-31 07:31:23,664 INFO werkzeug Thread-7 : 172.20.0.1 - - [31/Jan/2022 07:31:23] "GET /user?name=Am&sort=full_name&page=1 HTTP/1.1" 200 -
2022-01-31 07:31:37,109 INFO werkzeug Thread-8 : 172.20.0.1 - - [31/Jan/2022 07:31:37] "GET /user?email=am&name&sort=full_name&page=1 HTTP/1.1" 200 -
2022-01-31 07:31:39,417 INFO werkzeug Thread-9 : 172.20.0.1 - - [31/Jan/2022 07:31:39] "GET /user?email=a&name&sort=full_name&page=1 HTTP/1.1" 200 -
2022-01-31 07:31:45,974 INFO werkzeug Thread-10 : 172.20.0.1 - - [31/Jan/2022 07:31:45] "GET /user?email=a&name&sort=full_name&page=1&dir=desc HTTP/1.1" 200 -
2022-01-31 07:31:49,901 INFO werkzeug Thread-11 : 172.20.0.1 - - [31/Jan/2022 07:31:49] "GET /user?email=a&name&sort=full_name&page=2&dir=desc HTTP/1.1" 200 -
2022-01-31 07:31:54,907 INFO werkzeug Thread-12 : 172.20.0.1 - - [31/Jan/2022 07:31:54] "GET /user?email=a&name&sort=full_name&page=3&dir=desc HTTP/1.1" 200 -
2022-01-31 07:32:09,495 ERROR app Thread-13 : Exception on /user [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 113, in users_get
    sort = '-'+sort
TypeError: can only concatenate str (not "NoneType") to str
2022-01-31 07:32:09,503 INFO werkzeug Thread-13 : 172.20.0.1 - - [31/Jan/2022 07:32:09] "[35m[1mGET /user?email=a&name&page=3&dir=desc HTTP/1.1[0m" 500 -
2022-01-31 07:32:22,220 INFO werkzeug Thread-14 : 172.20.0.1 - - [31/Jan/2022 07:32:22] "GET /user?email=a&name&page=3&dir HTTP/1.1" 200 -
2022-01-31 07:32:24,982 INFO werkzeug Thread-15 : 172.20.0.1 - - [31/Jan/2022 07:32:24] "GET /user?email=a&name&page=3 HTTP/1.1" 200 -
2022-01-31 07:32:55,459 INFO werkzeug Thread-16 : 172.20.0.1 - - [31/Jan/2022 07:32:55] "GET /user?email=a&name HTTP/1.1" 200 -
2022-01-31 07:35:20,526 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:35:20,529 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:35:48,961 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:35:48] "GET /user?email=a&name HTTP/1.1" 200 -
2022-01-31 07:35:58,980 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:35:58] "GET /user?email=a&name=C&page=1 HTTP/1.1" 200 -
2022-01-31 07:44:25,179 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:44:25,181 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:45:14,160 ERROR app Thread-1 : Exception on /student [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 1802, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.InvalidTextRepresentation: invalid input syntax for type integer: ""
LINE 3: WHERE students.student_id = '' AND (students.full_name LIKE ...
                                    ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 289, in students_get
    for student in students:
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/orm/query.py", line 2887, in __iter__
    return self._iter().__iter__()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/orm/query.py", line 2894, in _iter
    result = self.session.execute(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/orm/session.py", line 1692, in execute
    result = conn._execute_20(statement, params or {}, execution_options)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 1614, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/sql/elements.py", line 325, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 1481, in _execute_clauseelement
    ret = self._execute_context(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 1845, in _execute_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 2026, in _handle_dbapi_exception
    util.raise_(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 1802, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.DataError: (psycopg2.errors.InvalidTextRepresentation) invalid input syntax for type integer: ""
LINE 3: WHERE students.student_id = '' AND (students.full_name LIKE ...
                                    ^

[SQL: SELECT students.id AS students_id, students.full_name AS students_full_name, students.student_id AS students_student_id, students.school_id AS students_school_id, students.route_id AS students_route_id, students.user_id AS students_user_id 
FROM students 
WHERE students.student_id = %(student_id_1)s AND (students.full_name LIKE '%%' || %(full_name_1)s || '%%')]
[parameters: {'student_id_1': '', 'full_name_1': 'Alys'}]
(Background on this error at: https://sqlalche.me/e/14/9h9h)
2022-01-31 07:45:14,318 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:45:14] "[35m[1mGET /student?name=Alys HTTP/1.1[0m" 500 -
2022-01-31 07:46:26,347 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:46:26,348 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:46:31,137 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:46:31] "GET /student?name=Alys HTTP/1.1" 200 -
2022-01-31 07:46:35,253 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:46:35] "GET /student?name=Aly HTTP/1.1" 200 -
2022-01-31 07:47:08,181 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 07:47:08] "GET /student?name HTTP/1.1" 200 -
2022-01-31 07:48:11,233 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 07:48:11] "GET /student HTTP/1.1" 200 -
2022-01-31 07:48:41,774 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:48:41,776 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:48:53,878 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:48:53] "GET /student HTTP/1.1" 200 -
2022-01-31 07:49:51,572 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:49:51,583 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:49:52,228 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:49:52] "[31m[1mGET /student HTTP/1.1[0m" 401 -
2022-01-31 07:49:57,105 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:49:57] "POST /login HTTP/1.1" 200 -
2022-01-31 07:50:07,848 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 07:50:07] "GET /student HTTP/1.1" 200 -
2022-01-31 07:50:42,553 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 07:50:42] "GET /student HTTP/1.1" 200 -
2022-01-31 07:50:57,298 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:50:57,300 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:50:59,962 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:50:59] "GET /student HTTP/1.1" 200 -
2022-01-31 07:51:36,414 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:51:36,417 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:52:17,835 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:52:17] "GET /student HTTP/1.1" 200 -
2022-01-31 07:52:28,958 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:52:28] "GET /student?name=Al HTTP/1.1" 200 -
2022-01-31 07:52:32,108 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 07:52:32] "GET /student?name HTTP/1.1" 200 -
2022-01-31 07:52:36,060 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 07:52:36] "GET /student HTTP/1.1" 200 -
2022-01-31 07:53:39,946 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:53:39,967 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:54:12,080 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:54:12,082 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:54:14,539 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:54:14] "GET /student HTTP/1.1" 200 -
2022-01-31 07:54:24,202 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:54:24] "GET /student?id=32 HTTP/1.1" 200 -
2022-01-31 07:55:01,224 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:55:01,226 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:55:03,151 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:55:03] "GET /student?id=32 HTTP/1.1" 200 -
2022-01-31 07:55:06,440 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:55:06] "GET /student?id HTTP/1.1" 200 -
2022-01-31 07:55:10,391 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 07:55:10] "GET /student HTTP/1.1" 200 -
2022-01-31 07:55:13,168 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 07:55:13] "GET /student?name HTTP/1.1" 200 -
2022-01-31 07:55:41,347 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:55:41,349 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:55:42,672 ERROR app Thread-1 : Exception on /student [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 1802, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.InvalidTextRepresentation: invalid input syntax for type integer: ""
LINE 3: WHERE students.student_id = '' AND (students.full_name LIKE ...
                                    ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 288, in students_get
    for student in students:
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/orm/query.py", line 2887, in __iter__
    return self._iter().__iter__()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/orm/query.py", line 2894, in _iter
    result = self.session.execute(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/orm/session.py", line 1692, in execute
    result = conn._execute_20(statement, params or {}, execution_options)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 1614, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/sql/elements.py", line 325, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 1481, in _execute_clauseelement
    ret = self._execute_context(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 1845, in _execute_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 2026, in _handle_dbapi_exception
    util.raise_(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/base.py", line 1802, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.DataError: (psycopg2.errors.InvalidTextRepresentation) invalid input syntax for type integer: ""
LINE 3: WHERE students.student_id = '' AND (students.full_name LIKE ...
                                    ^

[SQL: SELECT students.id AS students_id, students.full_name AS students_full_name, students.student_id AS students_student_id, students.school_id AS students_school_id, students.route_id AS students_route_id, students.user_id AS students_user_id 
FROM students 
WHERE students.student_id = %(student_id_1)s AND (students.full_name LIKE '%%' || %(full_name_1)s || '%%')]
[parameters: {'student_id_1': '', 'full_name_1': ''}]
(Background on this error at: https://sqlalche.me/e/14/9h9h)
2022-01-31 07:55:42,691 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:55:42] "[35m[1mGET /student?name HTTP/1.1[0m" 500 -
2022-01-31 07:57:23,006 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:57:23,010 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:57:27,456 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:57:27] "GET /student?name HTTP/1.1" 200 -
2022-01-31 07:57:33,335 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:57:33] "GET /student?name&id HTTP/1.1" 200 -
2022-01-31 07:57:36,102 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 07:57:36] "GET /student?name&id=32 HTTP/1.1" 200 -
2022-01-31 07:57:38,601 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 07:57:38] "GET /student?name&id HTTP/1.1" 200 -
2022-01-31 07:57:41,050 INFO werkzeug Thread-5 : 172.20.0.1 - - [31/Jan/2022 07:57:41] "GET /student?name&id=34 HTTP/1.1" 200 -
2022-01-31 07:58:44,983 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:58:44,986 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:59:05,737 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 07:59:05,739 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-01-31 07:59:09,378 INFO werkzeug Thread-1 : 172.20.0.1 - - [31/Jan/2022 07:59:09] "GET /student?name&id=34 HTTP/1.1" 200 -
2022-01-31 07:59:48,455 INFO werkzeug Thread-2 : 172.20.0.1 - - [31/Jan/2022 07:59:48] "GET /student?name&id&sort=full_name HTTP/1.1" 200 -
2022-01-31 08:00:05,211 INFO werkzeug Thread-3 : 172.20.0.1 - - [31/Jan/2022 08:00:05] "GET /student?name&id&sort=student_id HTTP/1.1" 200 -
2022-01-31 08:00:12,734 INFO werkzeug Thread-4 : 172.20.0.1 - - [31/Jan/2022 08:00:12] "GET /student?name&id&sort=school_id HTTP/1.1" 200 -
2022-01-31 08:00:23,422 INFO werkzeug Thread-5 : 172.20.0.1 - - [31/Jan/2022 08:00:23] "GET /student?name&id&sort=full_name&dir=desc HTTP/1.1" 200 -
2022-01-31 08:00:28,110 INFO werkzeug Thread-6 : 172.20.0.1 - - [31/Jan/2022 08:00:28] "GET /student?name&id&sort=full_name&dir=asc HTTP/1.1" 200 -
2022-01-31 17:37:32,119 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-01-31 17:37:32,121 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-02-01 16:09:25,136 INFO werkzeug Thread-1 : 172.20.0.1 - - [01/Feb/2022 16:09:25] "[31m[1mGET /student HTTP/1.1[0m" 401 -
2022-02-01 16:09:29,991 INFO werkzeug Thread-2 : 172.20.0.1 - - [01/Feb/2022 16:09:29] "POST /login HTTP/1.1" 200 -
2022-02-01 16:09:39,767 INFO werkzeug Thread-3 : 172.20.0.1 - - [01/Feb/2022 16:09:39] "GET /student HTTP/1.1" 200 -
2022-02-01 16:09:47,599 INFO werkzeug Thread-4 : 172.20.0.1 - - [01/Feb/2022 16:09:47] "GET /student?name HTTP/1.1" 200 -
2022-02-01 16:11:20,020 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-02-01 16:11:20,023 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-02-01 16:12:08,795 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-02-01 16:12:08,799 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-02-01 16:12:14,246 ERROR app Thread-1 : Exception on /student [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/usr/local/lib/python3.8/dist-packages/flask_jwt_extended/view_decorators.py", line 127, in decorator
    return current_app.ensure_sync(fn)(*args, **kwargs)
  File "/app/app.py", line 274, in students_get
    record_num = len(base_query)
TypeError: object of type 'BaseQuery' has no len()
2022-02-01 16:12:14,291 INFO werkzeug Thread-1 : 172.20.0.1 - - [01/Feb/2022 16:12:14] "[35m[1mGET /student HTTP/1.1[0m" 500 -
2022-02-01 16:15:35,869 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-02-01 16:15:35,870 INFO werkzeug MainThread :  * Running on http://172.20.0.3:5000/ (Press CTRL+C to quit)
2022-02-01 16:18:39,555 INFO werkzeug Thread-1 : 172.20.0.1 - - [01/Feb/2022 16:18:39] "GET /student HTTP/1.1" 200 -
2022-02-02 02:16:37,239 WARNING werkzeug MainThread :  * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
2022-02-02 02:16:37,245 INFO werkzeug MainThread :  * Running on http://172.21.0.4:5000/ (Press CTRL+C to quit)
2022-02-02 07:22:44,516 DEBUG root ThreadPoolExecutor-0_0 : [9, 5, 3]
2022-02-02 07:22:44,630 DEBUG root ThreadPoolExecutor-0_0 : in here9
2022-02-02 07:22:44,636 DEBUG root ThreadPoolExecutor-0_0 : in here5
2022-02-02 07:22:44,638 DEBUG root ThreadPoolExecutor-0_0 : in here3
2022-02-02 07:25:15,031 DEBUG root ThreadPoolExecutor-0_0 : [9, 5, 3]
2022-02-02 07:25:15,089 DEBUG root ThreadPoolExecutor-0_0 : in here9
2022-02-02 07:25:15,103 DEBUG root ThreadPoolExecutor-0_0 : in here5
2022-02-02 07:25:15,122 DEBUG root ThreadPoolExecutor-0_0 : in here3
